pkg/segment/query/processor/dataprocessor.go

```diff
@@ -79,9 +79,7 @@ func (dp *DataProcessor) Fetch() (*iqr.IQR, error) {
 	for {
 		gotEOF := false
 		input, err := dp.getStreamInput()
-		if err == io.EOF {
-			gotEOF = true
-		} else if err != nil {
+		if err != nil && err != io.EOF {
 			return nil, utils.TeeErrorf("DP.Fetch: failed to fetch input: %v", err)
```

Comment:
Because of the EOF returned from getStreamInput, even if the data processor does not return an EOF, we would exit the fetch because when we getEOF we return `output` and `io.EOF` over [here](https://github.com/siglens/siglens/blob/b1384e1783390c93b0e1a9f65efd261900dad724/pkg/segment/query/processor/dataprocessor.go#L102).